godot-core/src/builtin/vectors/vector3.rs

```diff
@@ -291,6 +292,32 @@ impl Vector3 {
         }
     }
 
+    /// Returns the spherical linear interpolation between the vector and `to` by the `weight` amount.
+    ///
+    /// The variable `weight` is representing the amount of interpolation, which is on the range of
+    /// 0.0 to 1.0.
+    ///
+    /// Length is also interpolated in the case that the input vectors have different lengths. If both
+    /// input vectors have zero length or are collinear to each other, the method instead behaves like
+    /// and directly calls [`Vector3::lerp`].
+    pub fn slerp(self, to: Self, weight: real) -> Self {
+        let start_length_sq: real = self.length_squared();
+        let end_length_sq = to.length_squared();
+        if start_length_sq == 0.0 || end_length_sq == 0.0 {
+            return self.lerp(to, weight);
+        }
+        let axis = self.cross(to);
+        let axis_length_sq: real = axis.length_squared();
+        if axis_length_sq == 0.0 {
+            return self.lerp(to, weight);
+        }
+        let unit_axis = axis.normalized();
+        let start_length = start_length_sq.sqrt();
+        let result_length = lerp(start_length, end_length_sq.sqrt(), weight);
+        let angle = self.angle_to(to);
+        self.rotated(unit_axis, angle * weight) * (result_length / start_length)
```

Comment:
Nitpick: 2x empty line after `}`
```suggestion

        let axis = self.cross(to);
        let axis_length_sq: real = axis.length_squared();
        if axis_length_sq == 0.0 {
            return self.lerp(to, weight);
        }
        
        let unit_axis = axis.normalized();
        let start_length = start_length_sq.sqrt();
        let result_length = lerp(start_length, end_length_sq.sqrt(), weight);
        let angle = self.angle_to(to);
        self.rotated(unit_axis, angle * weight) * (result_length / start_length)
```